src/mb_enhanced_cover_art_uploads/providers/qobuz.ts

```diff
@@ -1,29 +1,126 @@
 import { assertHasValue } from '../../lib/util/assert';
+import { gmxhr } from '../../lib/util/xhr';
 
 import type { CoverArt, CoverArtProvider } from './base';
 import { ArtworkTypeIDs } from './base';
 
-const ID_MATCH_REGEX = /\/album\/[^/]+\/([A-Za-z0-9]+)(?:\/|$)/;
+// Splitting these regexps up for each domain. www.qobuz.com includes the album
+// title in the URL, open.qobuz.com does not. Although we could make the album
+// title part optional and match both domains with the same regexp, this could
+// lead to issues with URLs like this:
+// https://open.qobuz.com/album/1234567890/related
+// Not sure if such URLs would ever occur, but using a single regexp could
+// lead to `related` being matched as the ID and the actual ID as the title.
+const WWW_ID_MATCH_REGEX = /\/album\/[^/]+\/([A-Za-z0-9]+)(?:\/|$)/;
+const OPEN_ID_MATCH_REGEX = /\/album\/([A-Za-z0-9]+)(?:\/|$)/;
+// Assuming this doesn't change often. If it does, we might have to extract it
+// from the JS code loaded on open.qobuz.com, but for simplicity's sake, let's
+// just use a constant app ID first.
+const QOBUZ_APP_ID = 712109809;
```

Comment:
I can confirm that this ID is not somehow user-dependent, I could extract the same number. Other than that there is not much to test for me, since I only know the one release with a booklet that you have probably already used for your test.

Reply:
Since the ID is embedded in the `main.js` file, I'm assuming it's neither user-dependent or time-dependent. But if they update their JS at some point, the App ID might change with it. Time will have to tell, which is why I'm forcing the provider to fail whenever the API doesn't return a valid response.

Reply:
That ID has been present in their `main.js` since 2017, according to [WBM](https://web.archive.org/web/20171230102128/http://open.qobuz.com/resources/0.0.28/js/main.js). So I doubt it'll stop working any time soon :)